nvflare/app_common/workflows/scatter_and_gather.py

```diff
@@ -171,6 +172,27 @@ def start_controller(self, fl_ctx: FLContext) -> None:
         fl_ctx.set_prop(AppConstants.START_ROUND, self._start_round, private=True, sticky=True)
         fl_ctx.set_prop(AppConstants.NUM_ROUNDS, self._num_rounds, private=True, sticky=False)
         self._global_weights = self.persistor.load(fl_ctx)
+        if not isinstance(self._global_weights, ModelLearnable):
+            self.system_panic(
+                reason=f"Expected global weights to be of type `ModelLearnable` but received {type(ModelLearnable)}",
+                fl_ctx=fl_ctx,
+            )
+            return
+
+        if self._global_weights.is_empty():
+            # see whether it is available from fl_ctx
+            self._global_weights = fl_ctx.get_prop(AppConstants.GLOBAL_MODEL)
+            if not isinstance(self._global_weights, ModelLearnable):
+                self.system_panic(
+                    reason=f"Expected global weights to be of type `ModelLearnable` but received {type(ModelLearnable)}",
+                    fl_ctx=fl_ctx,
+                )
+                return
+
+        if self._global_weights.is_empty():
+            self.system_panic(reason="Missing initial global model weights", fl_ctx=fl_ctx)
+            return
+
         fl_ctx.set_prop(AppConstants.GLOBAL_MODEL, self._global_weights, private=True, sticky=False)
```

Comment:
Please change to sticky=True to avoid the warning.

Reply:
changed. The warnings are gone.